proto/transport.proto

```diff
@@ -0,0 +1,224 @@
+syntax = "proto3";
+
+package smartcore.bos;
+
+option go_package = "github.com/vanti-dev/sc-bos/pkg/gen";
+
+import "google/protobuf/field_mask.proto";
+import "google/protobuf/timestamp.proto";
+import "types/info.proto";
+
+// TransportAPI represents something that can transport something else between places.
+// The transport will often be an elevator, but could be an escalator, conveyor etc.
+service TransportApi {
+  rpc GetTransport(GetTransportRequest) returns (Transport);
+  rpc PullTransport(PullTransportRequest) returns (stream PullTransportResponse);
+}
+
+service TransportInfo {
+  rpc DescribeTransport(DescribeTransportRequest) returns (TransportSupport);
+}
+
+// Transport describes the current state of the transport.
+message Transport {
+    enum Direction {
+      DIRECTION_UNSPECIFIED = 0;
+      // i.e. the device is stationary and not moving in any direction.
+      NO_DIRECTION = 1;
+      UP = 2;
+      DOWN = 3;
+      IN = 4;
+      OUT = 5;
+      CLOCKWISE = 6;
+      ANTICLOCKWISE = 7;
+      FORWARD = 8;
+      BACKWARD = 9;
+      EAST = 10;
+      WEST = 11;
+      NORTH = 12;
+      SOUTH = 13;
+    }
+
+    enum DoorStatus {
+      DOOR_STATUS_UNSPECIFIED = 0;
+      CLOSED = 1;
+      OPEN = 2;
+      OPENING = 3;
+      CLOSING = 4;
+      // The door is safety locked.
+      SAFETY_LOCKED = 5;
+      // The door remains in a position that is not fully open or closed.
+      LIMITED_OPENED = 6;
+    }
+
+    enum Mode {
```

Comment:
Is there any way we can avoid `Mode` as a name here, we already have the `mode` trait and `Electric.Mode` which gets confusing enough as it is. `OperationMode` maybe?